packages/builder-vite/vite-config.ts

```diff
@@ -81,19 +81,23 @@ export async function pluginConfig(options: ExtendedOptions, _type: PluginConfig
       },
     },
   ] as Plugin[];
-  if (framework === 'vue' || framework === 'vue3') {
+  if (/^vue3?$/.test(framework)) {
+    const isVue3 = framework === 'vue3';
     try {
-      const vuePlugin = require('@vitejs/plugin-vue');
+      const vuePlugin = isVue3
+        ? require('@vitejs/plugin-vue')
+        : require('@vitejs/plugin-vue2');
```

Comment:
maybe we can support vue < 2.7 here too? We just need to check vue version and require another plugin (vite-plugin-vue2)

Reply:
Hi @heykc, what do you think of this?

Reply:
Hey @Djaler / @IanVS, sorry I totally forgot to circle back to this.

I think this makes sense! Although after thinking about this a little, would it be so bad to have the user install the correct package themselves within viteFinal? We could just put a bit of documentation, probably in the README, depicting which vite plugin to use based on which vue plugin and vue version, and we can default to `@vitejs/plugin-vue` here. All the user would have to do is spread all existing plugins then have their correct vitejs plugin after

Reply:
I think I'm fine with that.  @Djaler you've got more experience with running / supporting vue 2.  What's your feeling on this idea?

Reply:
Personally, I prefer this plugin to support all three variations out of the box. I don't think we'll see more vue/vite-vue-plugin versions, so this setup will be stable for a long time